### ISSUE TITLE
`no-heading-trailing-punctuation`: reports puncuation between second to last and last code blocks

### ISSUE THREAD

**Issue by andreww2012** [ota-meshi/eslint-plugin-markdown-preferences#233]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.39.2

### What version of `eslint-plugin-markdown-preferences` are you using?

0.40.1

### What did you do?

This code should be allowed, but the last (second) comma is reported by this rule:

```md
# `foo`, `bar`, `baz`
```

### What did you expect to happen?

This code should be allowed.

### What actually happened?

The last comma is reported.

### Link to **GitHub Repo** with Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNp9Us2O2yAQfpUpvSRS7Eg9utqqUttzew+RwsLYJWsGC3Cb3SjvvgO290/ZlSyw4PubYc4iBr3Fk3JDj7UzohGf4dB6f9jA4VaFaXs4iI3A2FtKtfbU2q52x8hY6wYfEpzBYGsJf5Q7uEAbvAM5U7YTRYqvkrZbmDlOhTvj/9OC/T6Dl/MCf4P9E7DFgKQxvraohn7sLFULsBqekUWII8QEYezxF6VgmX8Dv2+PqFON08HqikmdCXFdt7ZPGFaSAFa7TZHZr+HmG3zKv7XDpGqD7KlVQrORtGZLPJXs3Bk19mV/6tBql6XOeQHAU0IysYHdtQRT7zgJau8cA9Hs2SATe0XdqDpsuA3XCudnzcpoqnhPSZ2kmImlqmbxh/fo5Ku/qIylrkpBWe5zVw0j6TSqZD0/UANfJolLEc7rPlfOwzIofcfR6mP0xHNSrKRI9wNmGht6wyHmRFIY/PcThxyWNL9Fxszprg1GFui50zE9lbQMwoeXH0/JS2IpRxJ/4vIIFZEZgA==
